ibodata/readers.py

```diff
@@ -0,0 +1,107 @@
+import os
+from datetime import datetime
+
+import numpy as np
+
+from ibodata.lateral_profile import LateralProfile
+from ibodata.depth_profile import DepthProfile
+
+
+def remove_incorrect_values(array_x_y):
+    """
+    Remove rows with incorrect values, such as Inf, -Inf, NaN, from given array
+    :param array_x_y: 2D array with 2 columns
+    :return: array without incorrect values
+    """
+    ret = np.asarray([array_x_y[:, 0][np.isfinite(array_x_y[:, -1])],
+                      array_x_y[:, -1][np.isfinite(array_x_y[:, -1])]]).transpose()
+    return ret
+
+
+def identify_profile_type(axis):
+    """
+    :param axis: name of axis
+    :return: 'L' for LateralProfile, 'D' for DepthProfile
+    :raises: Exception if couldn't identify profile type
+    """
+    if axis == 'PB' or axis == 'Z':
+        return 'D'
+    elif axis == 'X' or axis == 'Y':
+        return 'L'
+
+    raise Exception("Can't identify profile type")
+
+
+def get_axis(cvs_file_data):
+    """
+    :param cvs_file_data: data from file with named axis
+    :return: name of axis
+    """
+    axis_candidate_list = []
+    possible_axis = []
+    for axis in ('X', 'Y', 'Z', 'PB'):
+        if np.isin(axis, cvs_file_data.dtype.names):
+            possible_axis.append(axis)
+    for axis in possible_axis:
+        if np.all(np.diff(cvs_file_data[axis]) != 0):
+            axis_candidate_list.append(axis)
+    if len(axis_candidate_list) == 1:
+        return axis_candidate_list[0]
+    else:
+        raise Exception("Cannot find axis, candidates: " + str(axis_candidate_list))
+
+
+class ProfileDataset:
+    def __init__(self, profile, date):
+        self.profile = profile
+        self.date = date
+
+
+class FileReader:
+    def __init__(self, source):
+        """
+        :param source: directory path which will be searched
+        """
+        self.Lprofiles = list()
+        self.Dprofiles = list()
+
+        for dirname, dirnames, filenames in os.walk(source):
+            for filename in filenames:
+
+                # find file with .dat extension
+                if filename[-4:] == '.dat':
+
+                    # extract date from directory name
+                    split = dirname.split('\\')
```

Comment:
Can't this be done with `os.path.basename(path)` or some other method from here: https://docs.python.org/2/library/os.path.html ?